validator/server_arb/execution_run.go

```diff
@@ -79,6 +83,119 @@ func (e *executionRun) GetStepAt(position uint64) containers.PromiseInterface[*v
 	})
 }
 
+func (e *executionRun) GetMachineHashesWithStepSize(fromBatch, machineStartIndex, stepSize, numDesiredLeaves uint64) containers.PromiseInterface[[]common.Hash] {
+	return stopwaiter.LaunchPromiseThread(e, func(ctx context.Context) ([]common.Hash, error) {
+		return e.machineHashesWithStepSize(ctx, machineHashesWithStepSizeArgs{
+			startIndex:        machineStartIndex,
+			fromBatch:         fromBatch,
+			stepSize:          stepSize,
+			requiredNumHashes: numDesiredLeaves,
```

Comment:
* I don't really like arg structs unless there is a good reason, and here it's just for packing args that were just unpacked ( though this is a matter of taste)
* If going for an args_struct, I really don't like passing around functions in that streuct.. functions are more dangerous than most parameters
* It seems like you have forgotten to set a value for getMachineAtIndex?